scipy/signal/_spectral_py.py

```diff
@@ -586,12 +586,13 @@ def csd(x, y, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
         if Pxy.shape[-1] > 1:
             if average == 'median':
                 # np.median must be passed real arrays for the desired result
+                bias = _median_bias(Pxy.shape[-1])
                 if np.iscomplexobj(Pxy):
                     Pxy = (np.median(np.real(Pxy), axis=-1)
                            + 1j * np.median(np.imag(Pxy), axis=-1))
-                    Pxy /= _median_bias(Pxy.shape[-1])
```

Comment:
I think this is the difference @ilayn. Before the `bias` was computed with a modified Pxy. With the new version the `bias` is the same for both cases.

Reply:
I think the shape is constant no?

Reply:
From the description I think not, seems like sometimes x and y are permuted which would mean that shape[-1] varies.

Reply:
The behavior of "else"-block is unchanged.  But the behavior when "Pxy" is complex is changed.
The previous behanvior took `Pxy.shape[-1]` after median is applied, that means that it is actually taking `Pxy_original.shape[-2]` where `Pxy_original` is `Pxy` before median is computed.
This `_median_bias` should be computed with the number of frames, that is `Pxy_original.shape[-1]`.
So, this fix first compute `bias` before the if-statement, and applies it afterwards.

Reply:
Ah OK this is overwriting as it goes my bad sorry for the noise.

Reply:
No problem, was good to double check 👍